ws-server/src/types.rs

```diff
@@ -0,0 +1,83 @@
+use beef::lean::Cow;
+use serde::de::{self, Deserializer, Unexpected, Visitor};
+use serde::ser::Serializer;
+use serde::{Deserialize, Serialize};
+use serde_json::value::RawValue;
+use std::fmt;
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct JsonRpcRequest<'a> {
+	pub jsonrpc: TwoPointZero,
+
+	#[serde(borrow)]
+	pub id: Option<&'a RawValue>,
```

Comment:
technically not really to constrained to be a valid `request id` but I guess that's taken care of somewhere else to avoid allocations.

Reply:
Yup, if we get garbage in, we give garbage back, the id isn't really used anywhere internally (not in any map) so it doesn't really matter that we are more permissive than the spec.

The only thing that can happen is that we will return `null` on responses when request id was omitted.